compiler/rustc_span/src/lib.rs

```diff
@@ -1331,6 +1331,38 @@ impl SourceFileHash {
     }
 }
 
+#[derive(Copy, PartialEq, PartialOrd, Clone, Ord, Eq, Hash, Debug, Encodable, Decodable)]
+pub enum DebuggerVisualizerType {
+    Natvis,
+}
+
+rustc_data_structures::impl_stable_hash_via_hash!(DebuggerVisualizerType);
+
+/// A single debugger visualizer file.
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Encodable, Decodable)]
+pub struct DebuggerVisualizerFile {
+    /// The name of the debugger visualizer file.
+    pub name: FileName,
```

Comment:
Let's remove the `name` field. It does not seem to be used anywhere and it can only cause trouble around privacy if we encode file paths in crate metadata.

Reply:
@michaelwoerister  I have a related question here in regards to privacy. Would generating the natvis file in the temp directory and embedding it in the PDB also lead to a privacy concern? The debugger will show the full path of the Natvis file that was embedded in the PDB, whether that file still exists or not.

Reply:
That's a good question. Absolute paths are usually to be avoided because they can contain things like usernames. We'll have to take a closer look, but my guess is that NatVis files are not the only thing that will end up with an absolute path in the PDB.

Reply:
Do we know if the linker will embed an absolute path in the PDB even if it was passed a relative path in `/NATVIS:`?

Reply:
No, the linker will embed the path that was passed to the `/NATVIS:` flag. If it was passed a relative path, that's all that would be included.

Reply:
I did a quick experiment to see how rustc currently embeds the Natvis files for the standard library. It passes the absolute path of the Natvis files that live under the sysroot, which would include the username. So, this feature would not be adding any new privacy concerns since this is how it currently works today.